drivers/rtc/rtc_rv3028.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 ANITRA system s.r.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_RTC_RTC_RV3028_RTC_RV3028_H_
+#define ZEPHYR_DRIVERS_RTC_RTC_RV3028_RTC_RV3028_H_
+
+#include <zephyr/device.h>
+#include <sys/types.h>
+
+void rv3028_lock_sem(const struct device *dev);
+
+void rv3028_unlock_sem(const struct device *dev);
+
+int rv3028_read_regs(const struct device *dev, uint8_t addr, void *buf, size_t len);
+
+int rv3028_read_reg8(const struct device *dev, uint8_t addr, uint8_t *val);
+
+int rv3028_write_regs(const struct device *dev, uint8_t addr, const void *buf, size_t len);
+
+int rv3028_write_reg8(const struct device *dev, uint8_t addr, uint8_t val);
+
+int rv3028_update_reg8(const struct device *dev, uint8_t addr, uint8_t mask, uint8_t val);
+
+int rv3028_eeprom_wait_busy(const struct device *dev, int poll_ms);
+
+int rv3028_exit_eerd(const struct device *dev);
+
+int rv3028_enter_eerd(const struct device *dev);
+
+int rv3028_eeprom_command(const struct device *dev, uint8_t command);
```

Comment:
bit confused by this, didn't we agree to do this with function pointers?

Reply:
could you provide an example of what you mean?

Reply:
like how any driver works, sub-drivers should get a function pointer from the parent device if they need to access these functions and in a private header. That's what was agreed https://github.com/zephyrproject-rtos/zephyr/pull/74034#discussion_r1639551300

Reply:
I meant a code example... For a device driver specific header, there is no need for function pointers, the functions are not generic/interchangeable, so why add the complexity of a table of function pointers? (again, guessing what you mean by function pointers...)

Reply:
Because these are private functions, they should not be user-callable, they should only be implemented by the base driver and called by the other driver classes using this base driver

Reply:
I would really appreciate a code example of this. I'm not exactly sure how to declare a function pointer table without making it callable by the user.

Reply:
Regarding "user callable" this header is private, its in `drivers/rtc`, and included like `#include "rtc_rv3028"`. Its not in `include/zephyr/drivers/rtc`, it has to be included manually through CMake, hence `zephyr_library_include_directories(${ZEPHYR_BASE}/drivers/rtc)`

Reply:
I missed that it was in that folder and not in includes, solution is acceptable